### DISCUSSION TITLE
code was overwritten when pushed

### DISCUSSION THREAD

**Discussion by Ronneback** [community/community#53255] [Repositories]
I did:
git init
git add .
git commit -m "Add existing project files to Git"
git remote add origin https://github.com/user*/rep*
git push

I can't find any changes in my repository - It did no change ("your last chnage was 3 weeks ago").
I dot see what when wrong and where the code went?

I was able to recover it with [git reflog] but I don't see why this happened ?

**Comment by cohhi** [comment]
Find the project and branch that needs to be rolled back, click on History to view the history.
Find the version that needs to be rolled back and click Open.
Step 2: Rollback Operation
Click the Options Revert rollback button.
Select the branch that needs to be rolled back.
Step 3: Submit
ok

**Comment by airtower-luna** [comment]
Did the push actually work? The commands you listed would create a new local repository. If you try to push that to an existing remote repository, it should either fail (if you push a branch that exists in the remote) or end up on a new "orphan" (not connected to existing history) branch.

**Comment by Ronneback** [comment]
When I do [git checkout main] I got back to my old project, and when I do [git checkout backup] I have my updated project. How do I push my updated project to main without risking to overwrite it again?

**Comment by airtower-luna** [comment]
So there was no overwriting, you simply committed to different branches. Most likely that's a problem in itself, if I understood correctly the "new" code (the `backup` branch?) is an updated version of what's on `main`. In that case you should never have gone back to `git init`, and instead either committed to the existing branch, or created a new branch from the existing one (in that case you could just merge now).

**Reply by airtower-luna** [reply]
As you closed the thread, please don't forget to mark the answer. :slightly_smiling_face:
